CMakeLists.txt

```diff
@@ -132,9 +140,28 @@ ament_export_dependencies(
 
 if(BUILD_TESTING)
   find_package(ament_lint_auto REQUIRED)
+  find_package(test_msgs REQUIRED)
   ament_lint_auto_find_test_dependencies()
 
   add_subdirectory(test)
+
+  add_library(${PROJECT_NAME}_test_component_lib SHARED
+    test/${PROJECT_NAME}/test_component.cpp
+  )
+  ament_target_dependencies(${PROJECT_NAME}_test_component_lib
+    rclcpp
+    test_msgs
+  )
+  rclcpp_components_register_node(${PROJECT_NAME}_test_component_lib
+    PLUGIN "domain_bridge::TestComponent"
+    EXECUTABLE test_component)
+
+  install(TARGETS ${PROJECT_NAME}_test_component_lib
+    EXPORT export_${PROJECT_NAME}
+    ARCHIVE DESTINATION lib
+    LIBRARY DESTINATION lib
+    RUNTIME DESTINATION lib/${PROJECT_NAME}
+  )
```

Comment:
Installing the test library and using the same export name as the primary domain_bridge library causes issues with linking in downstream packages. For instance, I ran into a case where the dependency `test_msgs` could not be found when linking against the domain_bridge.

I don't think we need to install the library (or at the very least we can avoid exporting it).
See https://github.com/ros2/domain_bridge/pull/61 for a fix.